src/IECoreMaya/FromMayaMeshConverter.cpp

```diff
@@ -556,5 +643,24 @@ IECoreScene::PrimitivePtr FromMayaMeshConverter::doPrimitiveConversion( MFnMesh
 		}
 	}
 
+	bool convertCreases = creasesParameter()->getTypedValue();
+	if( convertCreases )
+	{
+		IntVectorData *cornerIds = new IntVectorData();
+		FloatVectorData *cornerSharpnesses = new FloatVectorData();
+
+		corners( cornerIds->writable(), cornerSharpnesses->writable() );
+
+		IntVectorData *creaseLengths = new IntVectorData();
+		IntVectorData *creaseIds = new IntVectorData();
+		FloatVectorData *creaseSharpnesses = new FloatVectorData();
+
+		creases( creaseLengths->writable(), creaseIds->writable(), creaseSharpnesses->writable() );
+
+		result->setCorners( cornerIds, cornerSharpnesses );
```

Comment:
There is some benefit to not calling `setCorners()` or `setCreases()` if the lists are empty, because the MeshPrimitive has optimisations for this case, whereby it won't need to copy/save/load an empty list around. It seems it'd be pretty easy to structure things to avoid even allocating the data for empty lists, if the `corners()` and `creases()` helpers allocated and assigned the data themselves, _after_ the `( !s )` early out.

Reply:
How's 2b866f8? I was thinking about making `creases()` and `corners()` free functions but I've now kept them in line with the existing methods...

Reply:
That's exactly what I had in mind, apart from the memory leaks ;)

Reply:
Fixed in 22e087b. Thanks!